unsuppressable error about invalid bitwise operation between different enumeration types

We use Infiniband API installed on hundreds of servers in our farm.  It is in /usr/include/infiniband/verbs.h.  For R&D, it is third-party software.
This program
`#include <infiniband/verbs.h>`
Then compilation fails like this
```
/grid/common/test/llvm-v18.1.1d1rh74_lnx86/bin/clang++ -c -std=c++26 -Wno-error -Wno-enum-conversion -Wno-deprecated-anon-enum-enum-conversion enum.cpp
In file included from enum.cpp:1:
/usr/include/infiniband/verbs.h:550:53: error: invalid bitwise operation between different enumeration types ('(unnamed enum at /usr/include/infiniband/verbs.h:539:1)' and 'ibv_create_cq_wc_flags')
  550 |         IBV_CREATE_CQ_SUP_WC_FLAGS = IBV_WC_STANDARD_FLAGS |
      |                                      ~~~~~~~~~~~~~~~~~~~~~ ^
  551 |                                 IBV_WC_EX_WITH_COMPLETION_TIMESTAMP |
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```
With -std=++2b, it does not.
The error is correct.  What is incorrect is that the error has no ID and cannot be suppressed in any known way.
According to this page
https://clang.llvm.org/docs/DiagnosticsReference.html#wanon-enum-enum-conversion
I can make the second enum anon, too, by adding -Dibv_create_cq_wc_flags= to the command line.  The error changes but stays an error:
```
/usr/include/infiniband/verbs.h:550:53: error: invalid bitwise operation between different enumeration types ('(unnamed enum at /usr/include/infiniband/verbs.h:539:1)' and
      '(unnamed enum at /usr/include/infiniband/verbs.h:524:1)')
```
This is a wrong general situation, every message must have an ID and be suppressable.
This is some info about the header:
-rw-r--r-- 1 root root 92862 Mar 22  2021 /usr/include/infiniband/verbs.h
a07809625d1024f1dfdf3db62a66ce88  /usr/include/infiniband/verbs.h
I am unsure if copying a couple of lines from that header to this ticket is a good idea.

--- a/clang/include/clang/Basic/DiagnosticSemaKinds.td
+++ b/clang/include/clang/Basic/DiagnosticSemaKinds.td
@@ -7639,9 +7639,13 @@ def warn_arith_conv_mixed_enum_types_cxx20 : Warning<
   "%sub{select_arith_conv_kind}0 "
   "different enumeration types%diff{ ($ and $)|}1,2 is deprecated">,
   InGroup<DeprecatedEnumEnumConversion>;
-def err_conv_mixed_enum_types_cxx26 : Error<
+
+def err_conv_mixed_enum_types: Error <
   "invalid %sub{select_arith_conv_kind}0 "
   "different enumeration types%diff{ ($ and $)|}1,2">;
+def warn_conv_mixed_enum_types_cxx26 : Warning <
+  err_conv_mixed_enum_types.Summary>,
+  InGroup<EnumEnumConversion>, DefaultError;
 
 def warn_arith_conv_mixed_anon_enum_types : Warning<
   warn_arith_conv_mixed_enum_types.Summary>,

--- a/clang/lib/Sema/SemaChecking.cpp
+++ b/clang/lib/Sema/SemaChecking.cpp
@@ -15323,7 +15323,7 @@ static bool checkBuiltinVectorMathMixedEnums(Sema &S, Expr *LHS, Expr *RHS,
            R = RHS->getEnumCoercedType(S.Context);
   if (L->isUnscopedEnumerationType() && R->isUnscopedEnumerationType() &&
       !S.Context.hasSameUnqualifiedType(L, R)) {
-    return S.Diag(Loc, diag::err_conv_mixed_enum_types_cxx26)
+    return S.Diag(Loc, diag::err_conv_mixed_enum_types)
            << LHS->getSourceRange() << RHS->getSourceRange()
            << /*Arithmetic Between*/ 0 << L << R;
   }

--- a/clang/lib/Sema/SemaExpr.cpp
+++ b/clang/lib/Sema/SemaExpr.cpp
@@ -1533,7 +1533,7 @@ void Sema::checkEnumArithmeticConversions(Expr *LHS, Expr *RHS,
     // In C++ 26, usual arithmetic conversions between 2 different enum types
     // are ill-formed.
     if (getLangOpts().CPlusPlus26)
-      DiagID = diag::err_conv_mixed_enum_types_cxx26;
+      DiagID = diag::warn_conv_mixed_enum_types_cxx26;
     else if (!L->castAs<EnumType>()->getDecl()->hasNameForLinkage() ||
              !R->castAs<EnumType>()->getDecl()->hasNameForLinkage()) {
       // If either enumeration type is unnamed, it's less likely that the

--- a/clang/test/SemaCXX/cxx2c-enum-compare.cpp
+++ b/clang/test/SemaCXX/cxx2c-enum-compare.cpp
@@ -1,9 +1,10 @@
-// RUN: %clang_cc1 %s -std=c++2c -fsyntax-only -verify -triple %itanium_abi_triple
+// RUN: %clang_cc1 %s -std=c++2c -fsyntax-only -verify=both,expected
+// RUN: %clang_cc1 %s -std=c++2c -fsyntax-only -verify=both,nowarn -Wno-enum-enum-conversion
 
 enum E1 { e };
 enum E2 { f };
 void test() {
-    int b = e <= 3.7; // expected-error {{invalid comparison of enumeration type 'E1' with floating-point type 'double'}}
+    int b = e <= 3.7; // both-error {{invalid comparison of enumeration type 'E1' with floating-point type 'double'}}
     int k = f - e; // expected-error {{invalid arithmetic between different enumeration types ('E2' and 'E1')}}
     int x = 1 ? e : f; // expected-error {{invalid conditional expression between different enumeration types ('E1' and 'E2')}}
 }

--- a/clang/test/SemaCXX/cxx2c-enum-compare.cpp
+++ b/clang/test/SemaCXX/cxx2c-enum-compare.cpp
@@ -1,5 +1,5 @@
 // RUN: %clang_cc1 %s -std=c++2c -fsyntax-only -verify=both,expected
-// RUN: %clang_cc1 %s -std=c++2c -fsyntax-only -verify=both,nowarn -Wno-enum-enum-conversion
+// RUN: %clang_cc1 %s -std=c++2c -fsyntax-only -verify=both -Wno-enum-enum-conversion
 
 enum E1 { e };
 enum E2 { f };

